### PR TITLE
AML-3550 AML-3552 AML-3528 AML-3553

### PR DESCRIPTION
**Pull request by Howesy77**

### CODE REVIEW

File: src/SFA.DAS.EmployerAccounts.Web.UnitTests/Orchestrators/EmployerAccountOrchestratorTests/WhenCreatingTheUserAccount.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+            _mediator.Setup(x => x.SendAsync(It.IsAny<CreateUserAccountCommand>()))
+                .ReturnsAsync(new CreateUserAccountCommandResponse()
+                {
+                    HashedAccountId = "ABS10"
+                });
+        }
+
+        [Test]
+        public async Task ThenTheMediatorCommandIsCalledWithCorrectParameters()
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.EmployerAccounts.Web.UnitTests/Orchestrators/EmployerAccountOrchestratorTests/WhenCreatingTheUserAccount.cs]: Can we make the test names less technical?  They're describing the implementation which isn't ideal.

(I realise you copied these from somewhere else! :))

File: src/SFA.DAS.EmployerAccounts.Web/Controllers/EmployerAccountController.cs [hunk 2]
```diff
@@ -136,25 +136,47 @@ public async Task<ActionResult> GateWayResponse()
...
+        [ValidateAntiForgeryToken]
+        [Route("getGovernmentFunding")]
+        public async Task<ActionResult> GetGovernmentFunding(int? choice)
+        {
+            switch (choice ?? 0)
             {
-                ViewBag.RequiresPayeScheme = string.IsNullOrEmpty(cookie.EmployerAccountPayeRefData.PayeReference) ||
-                                             cookie.EmployerAccountPayeRefData.EmpRefNotFound;
-            }
+                case 1:
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.EmployerAccounts.Web/Controllers/EmployerAccountController.cs]: Can the numbers be changed to an enum or constants so it's obvious what they're for?

File: src/SFA.DAS.EmployerAccounts/Commands/CreateUserAccount/CreateUserAccountCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+using SFA.DAS.HashingService;
+using SFA.DAS.NServiceBus;
+using SFA.DAS.Validation;
+using Entity = SFA.DAS.Audit.Types.Entity;
+
+
+namespace SFA.DAS.EmployerAccounts.Commands.CreateUserAccount
+{
+    //TODO this needs changing to be a facade and calling individual commands for each component
+    public class CreateUserAccountCommandHandler : IAsyncRequestHandler<CreateUserAccountCommand, CreateUserAccountCommandResponse>
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.EmployerAccounts/Commands/CreateUserAccount/CreateUserAccountCommandHandler.cs]: There is something I don't like about this class and it needs a significant refactor (it's doing way too much).  I realise it was a copy from somewhere else and now probably isn't the appropriate time to resolve it but it's worth having a chat about on Monday maybe?

Comment (Winsto) [reply src/SFA.DAS.EmployerAccounts/Commands/CreateUserAccount/CreateUserAccountCommandHandler.cs]: @MichaelYoung1981  Dan's  //TODO comment at the top of the originating class is from early 2017, so definitely overdue a refactor :0)

**Pull request merged**